src/ResultWriter/WaveFieldWriter.cpp

```diff
@@ -71,6 +71,60 @@ void seissol::writer::WaveFieldWriter::init(unsigned int numVars,
 	/** List of all buffer ids */
 	param.bufferIds[OUTPUT_PREFIX] = addSyncBuffer(m_outputPrefix.c_str(), m_outputPrefix.size()+1, true);
 
+	/** Extract the elements and vertices based on user given bounds */
+	// Reference to the vector containing all the elements
+	const std::vector<Element>& allElements = meshReader.getElements();
+
+	// Reference to the vector containing all the vertices
+	const std::vector<Vertex>& allVertices = meshReader.getVertices();
+
+	// Total number of elements
+	const size_t numTotalElems = meshReader.getElements().size();
+
+	// Elements of the extracted region
+	std::vector<const Element*> subElements;
+
+	// m_map will store a new map from new cell index to dof index
+	// the old map is contained in the "map" variable - which is a map from old
+	//    cell index to dof index
+	m_map = new unsigned int[numTotalElems];
+
+	// The oldToNewVertexMap defines a map between old vertex index to
+	// new vertex index. This is used to assign the vertex subset as well as
+	// used in MeshRefiner since the elements would hold old index of vertices
+	std::map<int, int> oldToNewVertexMap;
+
+
+	// Extract elements based on the region specified
+	for (size_t i = 0; i < numTotalElems; i++) {
```

Comment:
In case of "select all cells" I think it would be better to retain the old behaviour since I have changed the constructor for meshRefiner. With an if-statement for the loop and function overloading for the constructor this can be achieved.